### PR TITLE
Feature/display parks

### PR DESCRIPTION
**Pull request by baileydunning**
## Is this a fix or a feature?
Feature

## What is the change?
+ Fetch API data and set state of the `ParkContainer` and display unique info on page on load and by clicking a park button in the side navigation
+ Set up TypeScript interfaces to match the data of the API and pass those props to child components

## Where should the reviewer start?
+ `ParkContainer.tsx` & child components of `Banner` & `ParkInfo`
+ `App.tsx`

## How should this be tested?
+ See component dev tools when loading the page and by clicking on different park names
+ Reload the page to ensure a random park is generating 

## Relevant Tickets
+ #4 
+ #1
+ #5

### Gif

@saraho1123
@jaypeasee
@baileydunning

### CODE REVIEW

File: src/App/App.tsx [hunk 1]
```diff
@@ -1,16 +1,42 @@
...
+
+const App: React.FC = () => {
+  const [parkCode, setParkCode] = useState('')
+  
+  const choosePark = (parkCodeId: string) => {
+   setParkCode(parkCodeId)
+  }
+
+  const generateRandomParkCode = () => {
+    let index = Math.floor(Math.random() * nationalParks.length)
```
Comment (saraho1123) [inline src/App/App.tsx:25]: We get to use random again!

File: src/Banner/Banner.tsx [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
     return(
         <section>
-            <img 
+            {<img 
                 className="banner-img"
-                src="https://ewscripps.brightspotcdn.com/dims4/default/5a64170/2147483647/strip/true/crop/3000x1688+0+172/resize/1280x720!/quality/90/?url=https%3A%2F%2Fewscripps.brightspotcdn.com%2F0b%2F77%2F2d7dc37a4c59ba8a3f9e2709d4e9%2Fgrand-canyon-deaths-shutterstock-via-cnn-040519.JPG"
-                alt="Grand Canyon National Park"
-            />
-            <h1>Grand Canyon National Park</h1>
+                src={images[0].url}
```
Comment (saraho1123) [inline src/Banner/Banner.tsx:13]: Great work on getting this figured out Team!! We did awesome!

File: src/MainNav/MainNav.tsx [hunk 3]
```diff
@@ -3,7 +3,11 @@ import NavSearch from '../NavSearch/NavSearch'
 import ParkList from '../ParkList/ParkList'
 import parkfinderLogo from './landscape.png'
 
-function MainNav() {
+interface FuncProps {
```
Comment (saraho1123) [inline src/MainNav/MainNav.tsx:6]: Do we want to consider moving all the interfaces into one file? I'm not sure which is easier/better developer empathy? (Very much NOT a priority right now!)

File: src/ParkContainer/ParkContainer.tsx [hunk 4]
```diff
@@ -1,17 +1,38 @@
...
-            <Banner />
-            <UserNav />
-            <ParkInfo />
-        </section>
-        
-    )
+  return (
+    <section>
+      {currentPark && 
+        <Fragment>
```
Comment (saraho1123) [inline src/ParkContainer/ParkContainer.tsx:26]: Thanks for teaching me about Fragment!

**Review by saraho1123 — APPROVED** [review]
It is great working with both of you on this project! Thanks for sticking in there together and enjoying the struggle together!

File: src/ParkData.js [hunk 5]
```diff
@@ -3,440 +3,330 @@ export const nationalParks = [
     name: 'Acadia',
     state: 'ME',
     image: 'https://www.national-park.com/wp-content/uploads/2016/04/Welcome-to-Acadia-National-Park.jpeg',
-    activities: ['Arts and Culture', 'Cultural Demonstrations', 'Astronomy', 'Stargazing', 'Biking', 'Boating', 'Camping', 'Group Camping', 'Climbing', 'Rock Climbing', 'Compass and GPS', 'Geocaching', 'Fishing', 'Freshwater Fishing', 'Fly Fishing', 'Saltwater Fishing', 'Food', 'Picnicking', 'Guided Tours', 'Bus/Shuttle Guided Tour', 'Boat Tour', 'Hands-On', 'Citizen Science', 'Hiking', 'Front-Country Hiking', 'Horse Trekking', 'Horseback Riding', 'Ice Skating', 'Paddling', 'Canoeing', 'Kayaking', 'Stand Up Paddleboarding', 'Junior Ranger Program', 'Skiing', 'Cross-Country Skiing', 'Snow Play', 'Snowmobiling', 'Snowshoeing', 'Swimming', 'Freshwater Swimming', 'Saltwater Swimming', 'Wildlife Watching', 'Birdwatching', 'Park Film', 'Shopping', 'Bookstore and Park Store'],
```
Comment (jaypeasee) [inline src/ParkData.js:6]: Cleanup!

File: src/Banner/Banner.tsx [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
+type BannerProps = CurrentPark | CurrentParkContainer | Name | Images | ImageObject
 
 const Banner: React.FC<BannerProps> = props => {
-
+console.log(props)
     const { currentPark } = props as CurrentParkContainer
-    // const { currentPark } = props as CurrentParkContainer
-    // const { fullName } = props as Name
-    // const { url, altText } = props as Images
+    const { images } = currentPark as any
```
Comment (jaypeasee) [inline src/Banner/Banner.tsx:7]: We may want to change this "any" type to an array of objects later, but couldn't figure that out with an interface on this branch. "any" works for now.

**Comment by jaypeasee** [comment]
This is good to go! Good work team. Next up, router and user nav!

**Comment by jaypeasee** [comment]
Merging

File: src/Banner/Banner.tsx (hunk 6 above)
Comment (jaypeasee) [inline src/Banner/Banner.tsx:7]: We'll want to change this "any" to a more specific type or interface, but we had some trouble here. Put in the icebox for now.

**Pull request merged**